Imagine/Filter/PostProcessor/MozJpegPostProcessor.php

```diff
@@ -32,18 +32,24 @@ class MozJpegPostProcessor implements PostProcessorInterface, ConfigurablePostPr
     /** @var null|int Quality factor */
     protected $quality;
 
+    /** @var null|int quantization table */
+    protected $quantTable;
+
     /**
      * Constructor.
      *
      * @param string   $mozjpegBin Path to the mozjpeg cjpeg binary
      * @param int|null $quality    Quality factor
+     * @param int|null $quantTable    Quantization table
      */
     public function __construct(
```

Comment:
Wondering if there are any other ways we can go about this, it'd be nice to specify a global default and that's definitely functionality we should look to surface, but given the sheer number of options that we can pass cjpeg it feels like adding a them all to the constructor is going to get messy quickly.

I guess this is related to #909.  Should we go ahead with this and defer a nicer approach to 2.0, or should we look to solve this now?
cc @lsmith77 @robfrawley

Reply:
I guess the time is now for this “nicer” approach .. I guess we should use the OptionResolver component in place of ever extending constructor arguments?

Reply:
as such we I should add, as we are about to soon freeze 1.x for features I am ok adding “the last round” of constructor parameters (same for #909) .. and then immediately refactor to the OptionResolver for 2.x

wdyt @antoligy / @robfrawley ?

Reply:
I agree, the OptionsResolver is a great component and makes perfect sense everywhere we keep extending the constructor.